### DISCUSSION TITLE
micropython/lora debugging sx126x.py

### DISCUSSION THREAD

**Discussion by davefes** [micropython/micropython#13160] [Libraries & Drivers] [answered]
I am stuck not being able to load files above about 1KB on to a STM32WLE5CCU6 for the reason here:
https://github.com/orgs/micropython/discussions/13157
My plan was to put `sx126x.py` or `sx126.mpy` at `/flash` and change the `sys.path` order as suggested here:
https://github.com/orgs/micropython/discussions/13158
However, it looks like the NUCLEO_WL55 has a built-in programmer (ST-LinkV3).

Would a J-Link allow me to bring up a REPL and communicate with this board using rshell or mpremote?  Buying a ST-LinkV3 could come later.

**Comment by andrewleech** [comment]
The stlink works to provide repl by having a USB-uart built in, it comes up as multiple devices on PC and is wired to both swd and UART on the MCU.

If your jlink includes a separate UART/com port it could be connected the same, but requires the UART connection on MCU, it won't work over swd alone.

**Reply by davefes** [reply]
@andrewleech The stlink you refer to ... is it on the NUCLEO_WL55 dev board, a ST-LinkV3?  I only have a ST-LinkV2.   If I find a J-Link as you describe would flow-control still need to be sorted?

**Reply by andrewleech** [reply, marked as answer]
Flow control will still be the same issue, jlink/stlink com port channel used like this is no different to any other USB UART.

The nucleo wl55 is affected by this same issue, large file transfer won't work on it either.

**Reply by davefes** [reply]
OK.  I guess the fundamental question is  ... how do you develop Microython code (>1KB) on the wl55?  Must say, that using rshell/mpremote in an interactive development process ... without having to re-compile each time ... has been a real bonus.

After a year of working on ESP32 without an USB interface and big files I must have been lulled into a false sense of security!  They must handle their internal flash/UART comms in a different way.

I guess this little project goes on the back-burner until there is a break-thru.

**Comment by sophiedegran** [comment]
Dave, did you have a look on mischianti.org he have done a lot of work on lora for arduino and micropython on stm, e32, rp2 .. that's for ebyte card (driver sx126x as your) 
Check, hope that help you

**Reply by davefes** [reply]
@sophiedegran I have read through most of his stuff, I will re-read and try to find clues about the problem I am facing.  Thank you. 

Edit:  Dec 2023 is an interesting article. I have micropython/lora running on the BlackPill using a RFM96W LoRa chip.  I am now trying to get micropython/lora running on a E77-400M22S, but am facing a issue as this new part does not have an USB interface.

**Comment by davefes** [comment]
As mentioned in a previous post my project had gone on the back-burner.  Just would like to check if there has been any break-thru in the last two years.  Specifically, writing big files (>1-2KB) on a STM32WL55.
